### PR TITLE
Combat Screen Merge

### PR DESCRIPTION
**Pull request by owenrharding**
_The combat screen is built off of the NewScreen branch._

Contains new classes:
_screens_ package:
- CombatScreen
_components.combat_ package:
- CombatActions - handles exiting combat screen to main screen as well as event listeners
- CombatEnvironmentDisplay - displays background and will later include more 'environment' assets
- CombatExitDisplay - contains buttons to exit to the main screen. At the moment these are just 'win' or 'loss' conditions
- CombatStatsDisplay - displays attributes (health & attack) of the specific player and enemy entities involved in the collision
_areas.terrain_ package:
- CombatArea - doesn't really do much at the moment
_components_:
- EnterCombatComponent - based off 'EnterCombat' branch and handles calling a listener to create a new combat screen upon collision between and enemy entity and the player

Modifies these main classes:
**GdxGame** - mainly built off changes from the NewScreen branch, but includes overloads of functions which is required to pass enemy and player entities through to new instances of CombatScreen.
TouchAttackComponent - as collision is now handled in the 'EnterCombatComponent' class, TouchAttackComponent no longer needs to handle this.

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/Overlays/PauseDisplay.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                        logger.debug("Exit button clicked");
+                        eventService.globalEventHandler.trigger("exit");
+                    }
+                });
+
+        //button.setScale(0.75f);
+        //button2.setScale(0.75f);
+        // Layout buttons in a table
+        Table table = new Table();
+        table.add(button).size(button.getWidth() * 0.75f, button.getHeight() * 0.75f).center();
```
Comment (Jul-Wong) [inline source/core/src/main/com/csse3200/game/Overlays/PauseDisplay.java:106]: You could maybe consider a for loop for this

File: source/core/src/test/com/csse3200/game/combat/CombatScreenTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        this.container = new ServiceContainer(ServiceLocator.getEntityService(),
+                ServiceLocator.getRenderService(), ServiceLocator.getPhysicsService(),
+                ServiceLocator.getTimeSource(), ServiceLocator.getInputService(),
+                ServiceLocator.getResourceService(), ServiceLocator.getEventService());
+        this.player = PlayerFactory.createPlayer(game);
+        this.enemy = NPCFactory.createGhost(player);
+        this.combatScreen = new CombatScreen(game, oldScreen, container, player, enemy);
+    }
+
+    /*
```
Comment (Jul-Wong) [inline source/core/src/test/com/csse3200/game/combat/CombatScreenTest.java:40]: Are you testing with a video?

**Review by Jul-Wong — COMMENTED** [review]
Looks good to me. Just a couple minor things which I've commented on.

File: source/core/src/test/com/csse3200/game/combat/CombatScreenTest.java (hunk 2 above)
Comment (yessica-s) [reply source/core/src/test/com/csse3200/game/combat/CombatScreenTest.java:40]: We have some videos (screen recordings) of loading the combat screen from main screen. Although they are very informal and not yet uploaded anywhere

**Pull request closed without merging**